packages/react-dom/src/__tests__/ReactDOMServerIntegrationHooks-test.js

```diff
@@ -422,8 +427,13 @@ describe('ReactDOMServerHooks', () => {
         });
         return 'hi';
       }
-
-      const domNode = await render(<App />, 1);
+      const domNode = await render(
+        <App />,
+        render === clientRenderOnBadMarkup
```

Comment:
We use this pattern in other tests as well. E.g. `packages/react-dom/src/__tests__/ReactDOMServerIntegrationElements-test.js` makes extensive usage of assertions based on the renderer.

Reply:
Oh snap nice